picard/ui/scripteditor.py

```diff
@@ -0,0 +1,547 @@
+
+    def match_after_to_before(self):
+        """Sets the selected item in the 'after' list to the corresponding item in the 'before' list.
+        """
+        self.synchronize_selected_example_lines(self.current_row, self.ui.example_filename_before, self.ui.example_filename_after)
+
+    def match_before_to_after(self):
+        """Sets the selected item in the 'before' list to the corresponding item in the 'after' list.
+        """
+        self.synchronize_selected_example_lines(self.current_row, self.ui.example_filename_after, self.ui.example_filename_before)
+
+    def save_script(self):
+        """Emits a `save` signal to trigger appropriate save action in the parent object.
+        """
+        self.signal_save.emit()
+
+    def get_script(self):
+        """Provides the text of the file naming script currently loaded into the editor.
+
+        Returns:
+            str: File naming script
+        """
+        return str(self.ui.file_naming_format.toPlainText()).strip()
+
+    def set_script(self, script_text):
+        """Sets the text of the file naming script into the editor.  Sets default text if `script_text` is empty or missing.
+
+        Args:
+            script_text (str): File naming script text to set in the editor.
+        """
+        self.ui.file_naming_format.setPlainText(str(script_text).strip())
+
+    def update_example_files(self):
+        """Update the before and after file naming examples list.
+        """
+        self.examples.update_sample_example_files()
+        self.display_examples()
+
+    def update_examples(self):
+        """Update the before and after file naming examples using the current file naming script in the editor.
+        """
+        override = {'file_naming_format': self.get_script()}
+        self.examples.update_examples(override)
+        self.display_examples()
+
+    @staticmethod
+    def update_example_listboxes(before_listbox, after_listbox, examples):
+        before_listbox.clear()
+        after_listbox.clear()
+        for before, after in sorted(examples, key=lambda x: x[1]):
+            before_listbox.addItem(before)
+            after_listbox.addItem(after)
+
+    def display_examples(self, send_signal=True):
+        """Update the display of the before and after file naming examples.  Optionally emits an `update` signal.
+
+        Args:
+            send_signal (bool, optional): Determines if an `update` signal is emitted. Defaults to True.
+        """
+        self.current_row = -1
+        examples = self.examples.get_examples()
+        self.update_example_listboxes(self.ui.example_filename_before, self.ui.example_filename_after, examples)
+
+        if send_signal:
+            self.signal_update.emit()
+
+    def toggle_wordwrap(self):
+        """Toggles wordwrap in the script editing textbox.
+        """
+        if self.ui.file_naming_word_wrap.isChecked():
+            self.wordwrap = QtWidgets.QTextEdit.WidgetWidth
```

Comment:
What's the point of storing the value as a property? It isn't preserved anyway (perhaps it should be though).

Reply:
Removed for now.  I was thinking that perhaps some of the screen configuration settings should be preserved, but I was leaving that for a future PR.